### PR TITLE
add simple profiler, enabled by enable_profiler config

### PR DESCRIPTION
**Pull request by arvidn**
Analyze profilers with chia/utils/profiler.py.

What do people think about this? is it gross to make `chia/utils/profiler.py` also a tool usable as its own "main"?
If so, should we have a `tools` directory for things like this?

### CODE REVIEW

File: chia/full_node/full_node.py [hunk 1]
```diff
@@ -127,6 +128,10 @@ async def _start(self):
         self.mempool_manager = MempoolManager(self.coin_store, self.constants)
         self.weight_proof_handler = None
         asyncio.create_task(self.initialize_weight_proof())
+
+        if "enable_profiler" in self.config and self.config["enable_profiler"]:
```
Comment (Nikolaj-K) [inline chia/full_node/full_node.py:132]:
```python
if self.config("enable_profiler", False):
```

File: chia/util/profiler.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+
+if __name__ == "__main__":
+    import sys
+    import pstats
+    import io
+    from colorama import init, Fore, Back, Style
+    from subprocess import check_call
+
+    profile_dir = pathlib.Path(sys.argv[1])
```
Comment (Nikolaj-K) [inline chia/util/profiler.py:48]: fyi you can avoid the sys.argv list by using `argparse` and add variable names to what you read in.

File: chia/util/profiler.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            print(e)
+
+    def analyze_slot_range(profile_dir: pathlib.Path, first: int, last: int):
+        if last < first:
+            print("ERROR: first must be <= last when specifying slot range")
+            return
+
+        files = []
+        for i in range(first, last + 1):
+            files.append(str(profile_dir / ("slot-%05d.profile" % i)))
```
Comment (Nikolaj-K) [inline chia/util/profiler.py:119]: I think you want a function for this division that you use three times, and then you can comprehend the list as
```python
files = [the_function(idx) for idx in range(first, last + 1)]
```
or, depending on for whom this is written
```python
files = map(the_function, range(first, last + 1))
```

**Review by mariano54 — DISMISSED** [review]
LGTM, I think this would be useful in the wallet as well.

**Comment by arvidn** [comment]
yeah, I think it might make sense to extend this to cover the wallet and maybe other parts too, later. The directory the profiles are saved into would have to be parameterized to go into different ones.

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**